Design/DesignDoc.md

```diff
@@ -0,0 +1,220 @@
+    * If there is no data found, will throw a
+      `NoDataFoundException`
+
+## 6.4. GetTaggedExpense Endpoint
+
+* Accepts `GET` requests to `/expense/:tag`
+* Returns all the expenses for the requested tag in the ExpenseModel format.
+    * If there is no department found, will throw a
+      `InvalidDepartmentException`
+
+## 6.5. _UpdateExpense Endpoint_
+
+* Accepts `PUT` requests to `/expense/:date`
+* Accepts data to update an expense including an updated
+  expenseName, expenseAmount, tag, and date. Returns the updated expense.
+* If the expenseID or name is not found, will throw a   
+  `ExpenseNotFoundException`
+* For security concerns, we will validate the provided expense name does not
+  contain invalid characters: `" ' \`
+* If the expense name contains invalid characters, will throw an
+  `InvalidAttributeValueException`
+
+## 6.6. _AddExpense Endpoint_
+
+* Accepts `POST` requests to `/expense/:date`
+* Accepts data to create a new expenditure which includes the
+  expenseId, expenseName, expenseAmount, tag, and date. Returns the new   
+  expense.
+* For security concerns, we will validate the provided expense name does not
+  contain invalid characters: `" ' \`
+* If the expense name contains invalid characters, will throw an
+  `InvalidAttributeValueException`
+
+## 6.7. _GetBudget Endpoint_
+
+* Accepts `GET` requests to `/budget/:date`
+* Accepts a date and returns the corresponding BudgetModel.
+    * If the given date is not found, will throw a
+      `BudgetNotFoundException`
+
+## 6.6. _AddBudget Endpoint_
+
+* Accepts `POST` requests to `/budget/:date`
+* Accepts data to create a new budget which includes the
+  budgetId, targetAmount,and date. Returns the new   
+  budget.
+* For security concerns, we will validate the provided budget name does not
+  contain invalid characters: `" ' \`
+* If the budget name contains invalid characters, will throw an
+  `InvalidAttributeValueException`
+
+## 6.5. _UpdateBudget Endpoint_
+
+* Accepts `PUT` requests to `/budget/:date`
+* Accepts data to update a budget including an updated
+  targetAmount,and date. Returns the updated budget.
+* If the budgetID is not found, will throw a   
+  `BudgetNotFoundException`
+
+## 6.6. _BudgetStatus Endpoint_
+
+* Accepts `GET` requests to `/budget/:date`
+* Accepts a date and returns the status (achieved/ not achieved) of specified budget.
+  * If the given date is not found, will throw a
+    `BudgetNotFoundException`
+
+
+
+
+# 7. Tables
+
+_Define the DynamoDB tables you will need for the data your service will use. It may be helpful to first think of what objects your service will need, then translate that to a table structure, like with the *`Playlist` POJO* versus the `playlists` table in the Unit 3 project._
+
+* ExpenseTable
+  expenseId //  string
+  expenseName // string
+  expenseAmount // string
+  tag // string (GSI Partition Key)
+  date // partition key, string
```

Comment:
I think we need a sort key here as well. It's very likely that you'll have multiple expenses with the same date.